nodestream/pipeline/extractors/files.py

```diff
@@ -39,8 +40,12 @@ def read_file(self) -> Iterable[JsonLikeDocument]:
     @classmethod
     @contextmanager
     def open(cls, file: Path) -> "SupportedFileFormat":
-        with open(file, "r", encoding="utf-8") as fp:
-            yield cls.from_file_pointer_and_format(fp, file.suffix)
+        if (file and ".parquet" in file.name):
+            with open(file, "rb") as fp:
+                yield cls.from_file_pointer_and_format(fp, file.suffix)
+        else:
+            with open(file, "r", encoding="utf-8") as fp:
+                yield cls.from_file_pointer_and_format(fp, file.suffix)
 
```

Comment:
We recently landed https://github.com/nodestream-proj/nodestream/pull/264 which includes essentially an implementation on the proposal I made in the comments of this PR. 

This PR does need to be rebased against main, but I think we can remove if check while we are at it.